CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivityChains.uc

```diff
@@ -371,6 +373,44 @@ static function StateObjectReference FindRegionForFacilityChain ()
 	return EmptyRef;
 }
 
+static function X2DataTemplate CreateIntelInterceptionTemplate()
+{
+	local X2ActivityChainTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_IntelIntercept');
+	
+	Template.ChooseFaction = LastFaction;
+	Template.ChooseRegions = LastRegion;
+
+	Template.Stages.AddItem('Activity_IntelRescue');
+
+	return Template;
+}
+
+static function X2DataTemplate CreateSupplyInterceptionTemplate()
+{
+	local X2ActivityChainTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_SupplyIntercept');
+	
+	Template.ChooseFaction = LastFaction;
+	Template.ChooseRegions = LastRegion;
+
+	Template.Stages.AddItem('Activity_SupplyRescue');
+
+	return Template;
+}
+
+static function StateObjectReference LastFaction(XComGameState_ActivityChain ChainState, XComGameState NewGameState)
+{
+	return ChainState.ChainObjectRefs[0];
+}
+
+static function LastRegion(XComGameState_ActivityChain ChainState, out StateObjectReference PrimaryRegionRef, out StateObjectReference SecondaryRegionRef)
+{
+	PrimaryRegionRef = ChainState.ChainObjectRefs[1];
+}
```

Comment:
I feel like `LastFaction` and `LastRegion` are a very stupid way of doing this. Just unprotect the class vars in the chain state and set the values directly (and get rid of the delegates here)